### PR TITLE
Allow to use custom domain for b2 DownloadUrl #3653

### PR DESCRIPTION
**Pull request by aahung**
- how custom download url can be set with option "b2-download-url"

And the option will look like this:

image

Network-wise, the download url has been replaced as well:
image

### CODE REVIEW

**Comment by aahung** [comment]
#3653

File: Duplicati/Library/Backend/Backblaze/B2.cs [hunk 1]
```diff
@@ -163,6 +181,7 @@ public IList<ICommandLineArgument> SupportedCommands
                     new CommandLineArgument("auth-username", CommandLineArgument.ArgumentType.String, Strings.B2.AuthUsernameDescriptionShort, Strings.B2.AuthUsernameDescriptionLong),
                     new CommandLineArgument(B2_CREATE_BUCKET_TYPE_OPTION, CommandLineArgument.ArgumentType.String, Strings.B2.B2createbuckettypeDescriptionShort, Strings.B2.B2createbuckettypeDescriptionLong, DEFAULT_BUCKET_TYPE),
                     new CommandLineArgument(B2_PAGESIZE_OPTION, CommandLineArgument.ArgumentType.Integer, Strings.B2.B2pagesizeDescriptionShort, Strings.B2.B2pagesizeDescriptionLong, DEFAULT_PAGE_SIZE.ToString()),
+                    new CommandLineArgument(B2_DOWNLOAD_URL_OPTION, CommandLineArgument.ArgumentType.String, Strings.B2.B2downloadurlDescriptionShort, Strings.B2.B2downloadurlDescriptionLong),
```
Comment (Pectojin) [inline Duplicati/Library/Backend/Backblaze/B2.cs]: I would recommend using the overloaded `CommandLineArgument` method that allows defining default value.

Then you can delete the part of the long description that states the default value.

`public CommandLineArgument(string name, ArgumentType type, string shortDescription, string longDescription, string defaultValue, string[] aliases)
            : this(name, type, shortDescription, longDescription, defaultValue)`

Comment (aahung) [reply Duplicati/Library/Backend/Backblaze/B2.cs]: Actually there is no default hardcoded value for `downloadUrl`. I put it in the description as `https://f00X.backblazeb2.com` where `X` depends on the account. Maybe I need to clarify it in the description a little bit. Thanks

**Comment by aahung** [comment]
@Pectojin I revised the description in [49c7050](
https://github.com/duplicati/duplicati/pull/3654/commits/49c70503b8765eb7cc0df786617a49c3fd6927ae) to avoid the confusion.

File: Duplicati/Library/Backend/Backblaze/B2.cs (hunk 1 above)
Comment (Pectojin) [reply Duplicati/Library/Backend/Backblaze/B2.cs]: Oh, that makes more sense

**Comment by Pectojin** [comment]
Adding this parameter to the destination form when editing/creating a B2 backup may be a nice addition to this, but lets merge the base functionality as is.

**Pull request merged**